tests/unit_tests/test_tracks.py

```diff
@@ -0,0 +1,103 @@
+from pathlib import Path
+
+import numpy as np
+import openmc
+import pytest
+
+from tests.testing_harness import config
+
+
+@pytest.fixture
+def sphere_model():
+    openmc.reset_auto_ids()
+    mat = openmc.Material()
+    mat.add_nuclide('Zr90', 1.0)
+    mat.set_density('g/cm3', 1.0)
+
+    model = openmc.Model()
+    sph = openmc.Sphere(r=25.0, boundary_type='vacuum')
+    cell = openmc.Cell(fill=mat, region=-sph)
+    model.geometry = openmc.Geometry([cell])
+
+    model.settings.run_mode = 'fixed source'
+    model.settings.batches = 1
+    model.settings.particles = 100
+
+    return model
+
+
+def generate_track_file(model, **kwargs):
+    # If running in MPI mode, setup proper keyword arguments for run()
+    kwargs.setdefault('openmc_exec', config['exe'])
+    if config['mpi']:
+        kwargs['mpi_args'] = [config['mpiexec'], '-n', config['mpi_np']]
+    model.run(**kwargs)
+
+    if config['mpi'] and int(config['mpi_np']) > 1:
+        # With MPI, we need to combine track files
+        track_files = Path.cwd().glob('tracks_p*.h5')
+        openmc.TrackFile.combine(track_files, 'tracks.h5')
+    else:
+        track_file = Path('tracks.h5')
+        assert track_file.is_file()
+
+
+def test_tracks(sphere_model, run_in_tmpdir):
+    # Set track identifiers
+    sphere_model.settings.track = [(1, 1, 1), (1, 1, 10), (1, 1, 75)]
+
+    # Run OpenMC to generate tracks.h5 file
+    generate_track_file(sphere_model)
+
+    # Open track file and make sure we have correct number of tracks
+    tracks = openmc.TrackFile('tracks.h5')
+    assert len(tracks) == len(sphere_model.settings.track)
+
+    for track, identifier in zip(tracks, sphere_model.settings.track):
+        # Check attributes on Track object
+        assert isinstance(track, openmc.Track)
+        assert track.identifier == identifier
+        assert isinstance(track.particles, list)
+        assert len(track.particles) == 1
+
+        # Check attributes on ParticleTrack object
+        particle_track = track.particles[0]
+        assert isinstance(particle_track, openmc.ParticleTrack)
+        assert particle_track.particle == openmc.ParticleType.NEUTRON
```

Comment:
This just reminded me, should we run the track mode with a photon source and check the ```particletype``` is photon and the rest of the track file works as I think this feature might be popular for photon detectors.

Reply:
Good idea, I parameterized this test so that it runs for both neutrons and photon